src/can-be-asserted.js

```diff
@@ -1,10 +1,12 @@
 import _ from 'lodash';
+import React from 'react';
 
 export default function canBeAsserted(vdom) {
     return _.all([].concat(vdom), node =>
-        node
+        React.isValidElement(node) ||
+        (node
         && typeof node === 'object'
         && typeof node.type === 'string'
-        && (!node.props || typeof node.props === 'object')
+        && (!node.props || typeof node.props === 'object'))
```

Comment:
Nice
Maybe we should extract the entire right expression (after the `||`) into a local function, `isNativeElement`?